src/Controller/FrontPageConfigurationController.php

```diff
@@ -20,37 +22,92 @@
   * @IsGranted("ROLE_ADMIN")
   */
 final class FrontPageConfigurationController extends AbstractController {
-    public function frontPageConfig(ForumRepository $fr, SubmissionRepository $sr, ForumConfigurationRepository $fcr, EntityManager $em, Request $request) {
-        $data = new ForumConfigurationData($fcr->findSitewide());
 
-        $form = $this->createForm(ForumConfigurationType::class, $data);
+    public function frontPageConfig(
+        ForumRepository $fr,
+        SubmissionRepository $sr,
+        ForumConfigurationRepository $fcr,
+        UserRepository $ur,
+        EntityManager $em,
+        Request $request
+    ) {
+        $announcementForumName = "announcements";
+        $data = new NewForumAnnouncementData();
+
+        $form = $this->createForm(NewForumAnnouncementType::class, $data);
         $form->handleRequest($request);
 
         $message = "";
+        $messageClass = "";
 
         if ($form->isSubmitted() && $form->isValid()) {
             $fc = $data->toForumConfiguration();
 
-            // For sitewide settings, we need to force this to null.
-            $fc->setForumId(null);
+            // Find the user and forum for submission.
+            $user = $ur->loadUserByUsername($this->get('security.token_storage')->getToken()->getUser()->getUsername());
+            $forum = $fr->findOneByCaseInsensitiveName($announcementForumName);
+
+            if($forum != null) {
+                // Create the submission.
+                $submission = new Submission(
+                    $data->threadTitle,
+                    null,
+                    $data->threadContent,
+                    $forum,
+                    $user,
+                    null
+                );
+                $em->persist($submission);
+                $em->flush();
+
+                // Load the current announcement's row id. Sitewide announcemnts have null forum id.
+                $fc->setId($fcr->findSitewide()->getId());
+                $fc->setAnnouncementSubmissionId($submission->getId());
+                $fc->setForumId(null);
 
-            if($fc->getId() == null || trim($fc->getId()) == "")
-                $em->persist($fc);
-            else
-                $em->merge($fc);
+                if($fc->getId() == null || trim($fc->getId()) == "")
+                    $em->persist($fc);
+                else
+                    $em->merge($fc);
```

Comment:
This can be changed to be bit cleaner/readable
```
// get existing
$fc = $fcr->findSitewide()
if (is_null($fc->getId()) || trim($fc->getId()) == "") {
    // If not existing, create new one
    $fc = $data->toForumConfiguration();
}
// Set the new data
$fc->setForumId(null);
$fc->setAnnouncementSubmissionId($submission->getId());
$em->persist($fc);
```